test/test_google_cloud_manage.py

```diff
@@ -1071,6 +1076,40 @@ def test_service_account_keys_when_empty(test_cloud_manager):
         any(account in str(kwarg) for kwarg in kwargs.values())
     )
 
+def test_get_service_account_type(test_cloud_manager):
+
+    service_account = {
+        "email": "test@appspot.gserviceaccount.com"
+    }
+    test_cloud_manager._authed_session.get.return_value = (
+        _fake_response(200, service_account)
+    )
+    assert test_cloud_manager.get_service_account_type(service_account) == COMPUTE_ENGINE_DEFAULT_SERVICE_ACCOUNT
+
+    service_account = {
```

Comment:
let's break these into separate tests. sorry I didn't note this before. Any time we're testing some different senario (especically if we're changing the return value for something) we should split it into a different test. so we'll have like `test_get_compute_engine_service_account` and `test_get_google_api_service_account` etc.